### PR TITLE
Rule (java) anonymous inner classes containing only one method should become lambdas

### PR DESCRIPTION
**Pull request by formalhaut69**
### Description of the Change
Made a change to ValueStore.java to comply with this rule

### CODE REVIEW

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java [hunk 1]
```diff
@@ -67,41 +67,41 @@ public void setObject(final Object value) {
     }
 
     public final void save() {
-        converter.save();
+        converter.save(access);
     }
 
-    private static interface Saver extends Serializable {
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:73]: I wonder what performance/memory complications there are when we remove the `static` modifier. @formalhaut69 did this have to be done to make it a lambda or was it for another reason?

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java [hunk 2]
```diff
@@ -67,41 +67,41 @@ public void setObject(final Object value) {
...
-
-        public void save();
+    private interface Saver extends Serializable {
+        
+        public void save(ParameterWriteAccess PWA);
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:75]: Can these be a `final` parameter?

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java [hunk 3]
```diff
@@ -67,41 +67,41 @@ public void setObject(final Object value) {
...
     public final void save() {
-        converter.save();
+        converter.save(access);
     }
 
-    private static interface Saver extends Serializable {
-
-        public void save();
+    private interface Saver extends Serializable {
+
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:74]: A minor comment but when possible please review your NetBeans 11 settings and consider changing it to trim trailing whitespaces.

image

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java (hunk 1 above)
Comment (arcturus2) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:73]: > https://gitter.im/constellation-app/community?at=5e83ebd9b6d72077bcabfe17

@cygnus-x-1 I've just caught up on the chat in Gitter. Can you explain why having the `static` was a bad design?

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java (hunk 2 above)
Comment (formalhaut69) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:75]: Yeah, sorry I keep forgetting to do that

Comment (cygnus-x-1) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java]: Also an all uppercase variable name like that is not good Java style.

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java (hunk 1 above)
Comment (cygnus-x-1) [reply CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/locking/ValueStore.java:73]: > @cygnus-x-1 I've just caught up on the chat in Gitter. For my understanding, can you explain why having the `static` is bad design?

Having an inner static Interface like that is not inherently bad design (although it does imply it's supposed to be used outside this class); the bad design part was then having it rely on a variable of its outer class, as that variable will only exist when that outer class is instantiated.

**Comment by arcturus2** [comment]
@formalhaut69 I resolved a merge conflict I introduced merging another PR. Let me know if it looks OK.

@cygnus-x-1 and @antares1470 would you be able to review this PR?

**Pull request merged**